charms/istio-pilot/src/charm.py

```diff
@@ -141,14 +102,14 @@ def handle_default_gateway(self, event):
         t = self.env.get_template('gateway.yaml.j2')
         gateway = self.model.config['default-gateway']
         manifest = t.render(name=gateway, app_name=self.app.name)
-        self._delete_existing_resource_objects(
-            resource=self.gateway_resource,
+        self._resource_handler.delete_existing_resource_objects(
+            resource=self._custom_resources['gateway'],
             labels={
-                "app.juju.is/created-by": f"{self.app.name}",
                 "app.{self.app.name}.io/is-workload-entity": "true",
```

Comment:
This is a mistake from before this PR, but this string should be an f-string
```suggestion
                f"app.{self.app.name}.io/is-workload-entity": "true",
```

Is the `juju/created-by` label being removed because of redundancy?

Reply:
+1 on the typo, the juju label is a default in the methods that depend on labels inside `resource_handler.py`, that's because we always care about the resources that are created by juju.